src/tests/unit_tests/test_util.py

```diff
@@ -189,3 +194,95 @@ def test_extract_numbers_from_text_fails(source_data, incorrect_result_data):
     # test aganist a text which don't have the same numers as the source
     check_result = check_if_number_exist(incorrect_result_data, source_data)
     assert not check_result["number_exists"]
+
+def test_json_extractor_valid_json():
```

Comment:
nice tests.
these tests could also be made more concise with parametrize.

remember to check the failure cases and corner cases like `None` and `""`